### PR TITLE
Np 46846 serie journal lookup by name and issn

### PR DESCRIPTION
**Pull request by truhacevkir**
Implemented lookup for Journal and Serie by issn's and title as we do in brage-import.

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/PublishingChannelEntryResolver.java [hunk 1]
```diff
@@ -53,7 +57,24 @@ private Optional<? extends URI> lookupCristinPublisherByhName() {
     }
 
     public URI createJournal() {
-        return lookUpNsdJournal().orElse(persistChannelRegistryExceptionReport("Journal"));
+        return lookUpNsdJournal()
+                   .or(() -> lookupJournalByTitle().map(this::toJournalUri))
+                   .or(() -> lookupJournalByIssn().map(this::toJournalUri))
+                   .orElseGet(() -> persistChannelRegistryExceptionReport("Journal"));
+    }
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/PublishingChannelEntryResolver.java:64]: Making lookup by nsd code -> title -> issn

File: cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java [hunk 2]
```diff
@@ -523,6 +526,98 @@ void shouldCreateNullPublisherWhenAllPublisherDataIsMissing() {
         assertThat( book.getPublisher(), is(instanceOf(NullPublisher.class)));
     }
 
+    @Test
```
Comment (truhacevkir) [inline cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java:529]: Here are all the tests :)

File: cristin-import/src/test/resources/features/JournalArticleRules.feature [hunk 3]
```diff
@@ -21,15 +21,15 @@ Feature: Mapping of "Article in business/trade/industry journal", "Academic arti
 
   Scenario: Cristin Entry's Journal Publication "issn" entry is copied to the NVA field  "printISSN".
     Given that the Cristin Result has a non empty Journal Publication
-    And the Journal Publication has a "issn" entry equal to "1903-6523"
+    And the Journal Publication has a "issn" entry equal to "2434-561X"
```
Comment (truhacevkir) [inline cristin-import/src/test/resources/features/JournalArticleRules.feature:24]: Replacing issn with the one which does not exist in the csv file

File: cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java:529]: I sense that you are struggling with cucumber?

Comment (truhacevkir) [reply cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java:529]: Not struggling, but it takes a bit more time to implement them :)

Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/mapper/CristinMapperTest.java:529]: Siden det ikke er meg som skal leve med dette, så skal jeg ikke kreve av deg at du skriver det om som cucumber tester. Men si ifra hvis du ønsker å ha disse som cucumber tester, men du opplever det som tungt.

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**